### PR TITLE
Add Cost Insights to demo site

### PR DESCRIPTION
**Pull request by timbonicus**
Adds `@backstage/plugin-cost-insights` to the demo site. Upgrades all the @backstage things. Fixes a port conflict in the development app-config.

### CODE REVIEW

**Comment by timbonicus** [comment]
The port change is the only unrelated fix, for the dependencies I ran into some cascading dependency updates due to `plugin-cost-insights` dependencies and decided to go whole-hog. I could try to tease out which ones specifically needed to be bumped 🤷‍♂️

**Comment by Rugvip** [comment]
Yup, just doing the non-cost-insights things in a separate PR will clean this one up and be a nice thing to link to

File: packages/app/src/apis.ts [hunk 1]
```diff
@@ -7,6 +7,12 @@ import {
   configApiRef,
 } from '@backstage/core';
 
+import {
+  costInsightsApiRef,
+  ExampleCostInsightsClient,
+} from '@backstage/plugin-cost-insights';
+
+
```
Comment (freben) [inline packages/app/src/apis.ts:15]: Every extra newline kills a kitten

**Comment by timbonicus** [comment]
I reverted the unnecessary package updates. There's still quite a few needed to get a consistent `@backstage/core` due to https://github.com/backstage/backstage/issues/3185

File: packages/app/src/apis.ts (hunk 1 above)
Comment (timbonicus) [reply packages/app/src/apis.ts]: I think it's actually exponential kitten-killing as the number of extra newlines increases 🤔

**Review by Rugvip — APPROVED** [review]

**Comment by timbonicus** [comment]
Do we not have CI/CD setup for the demo site? I don't see this reflected in the deployed version.

**Pull request merged**